### PR TITLE
[build] Switch packageManager to pnpm

### PR DESCRIPTION
**Pull request by spacey-sooty**
Resolves https://github.com/SleipnirGroup/Choreo/issues/528

### CODE REVIEW

**Comment by calcmogul** [comment]
I guess pnpm isn't a drop-in replacement?
```
    Finished `release` profile [optimized] target(s) in 1m 54s
       Error failed to rename `/Users/runner/work/Choreo/Choreo/target/release/choreo` to `/Users/runner/work/Choreo/Choreo/target/release/Choreo`: No such file or directory (os error 2)
 ELIFECYCLE  Command failed with exit code 1.
```

File: .github/workflows/choreo.yml [hunk 1]
```diff
@@ -71,7 +75,7 @@ jobs:
         if: startsWith(matrix.os, 'macOS')
 
       - name: Build package
-        run: npm run tauri build ${{ matrix.tauri-build-flags }}
+        run: pnpm tauri build ${{ matrix.tauri-build-flags }}
```
Comment (calcmogul) [inline .github/workflows/choreo.yml:78]: Why does lint-format.yml use `pnpm run` while this doesn't?

Comment (spacey-sooty) [reply .github/workflows/choreo.yml:78]: 'Twas an attempt at debugging Mac and windows arm failures

**Comment by spacey-sooty** [comment]
> I guess pnpm isn't a drop-in replacement?
> 
> ```
> 
>     Finished `release` profile [optimized] target(s) in 1m 54s
> 
>        Error failed to rename `/Users/runner/work/Choreo/Choreo/target/release/choreo` to `/Users/runner/work/Choreo/Choreo/target/release/Choreo`: No such file or directory (os error 2)
> 
>  ELIFECYCLE  Command failed with exit code 1.
> 
> ```

Yeah I'd only tested on Linux x86 cause that's all I have access too and I haven't really been able to debug this issue due to not having a mac or windows computer.

**Comment by spacey-sooty** [comment]
This weird because pnpm is mentioned in Taurus docs as an option to use

**Comment by spacey-sooty** [comment]
The issue was pnpm ignores the first --

**Comment by spacey-sooty** [comment]
Some benchmarks for reference (using `time`)
pnpm without cache:
```
Done in 12.2s

real    0m12.633s
user    0m23.105s
sys     0m5.264s
```

pnpm with full cache and no changes
```
Done in 1.5s

real    0m1.766s
user    0m2.300s
sys     0m0.295s
```

npm without cache:
```
real    1m17.311s
user    0m41.967s
sys     0m8.806s
```

npm with full cache and no changes:
```
real    0m3.014s
user    0m2.955s
sys     0m0.299s
```

File: docs/contributing/building-choreo.md [hunk 2]
```diff
@@ -25,33 +25,33 @@
...
 ```console
-npm run tauri dev
+pnpm run tauri dev
 ```
 
 Start development server (release).
 
 ```console
-npm run tauri dev -- --release
+pnpm run tauri dev -- --release
```
Comment (calcmogul) [inline docs/contributing/building-choreo.md:54]:
```suggestion
pnpm run tauri dev --release
```

**Review by calcmogul — CHANGES_REQUESTED** [review]

**Review by calcmogul — APPROVED** [review]

**Pull request merged**